src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -1000,6 +1001,100 @@ class AlterDefaultPrivilegesRevokeSegment(BaseSegment):
+    """
+
+    type = "comment_on_statement"
+
+    match_grammar = StartsWith(Sequence("COMMENT", "ON"))
+    parse_grammar = Sequence(
+        "COMMENT",
+        "ON",
+        Sequence(
+            OneOf(
+                Sequence(
+                    OneOf(
+                        "TABLE",
+                        # TODO: Create a ViewReferenceSegment
+                        "VIEW",
+                    ),
+                    Ref("TableReferenceSegment"),
+                ),
+                Sequence(
+                    "COLUMN",
+                    # TODO: Does this correctly emit a Table Reference?
+                    Ref("ColumnReferenceSegment"),
+                ),
+                Sequence(
+                    "CONSTRAINT",
+                    Ref("ObjectReferenceSegment"),
+                    Sequence(
+                        "ON",
+                        Ref.keyword("DOMAIN", optional=True),
+                        Ref("ObjectReferenceSegment"),
+                    ),
+                ),
+                Sequence(
+                    "DATABASE",
+                    Ref("DatabaseReferenceSegment"),
+                ),
+                Sequence(
+                    "EXTENSION",
+                    Ref("ExtensionReferenceSegment"),
+                ),
+                Sequence(
+                    "FUNCTION",
+                    Ref("FunctionNameSegment"),
+                    Ref("FunctionParameterListGrammar"),
+                ),
+                Sequence(
+                    "INDEX",
+                    Ref("IndexReferenceSegment"),
+                ),
+                Sequence(
+                    "SCHEMA",
+                    Ref("SchemaReferenceSegment"),
+                ),
+                # TODO: Split out individual items if they have references
+                Sequence(
+                    OneOf(
+                        "COLLATION",
+                        "CONVERSION",
+                        "DOMAIN",
+                        "LANGUAGE",
+                        "POLICY",
+                        "PUBLICATION",
+                        "ROLE",
+                        "RULE",
+                        "SEQUENCE",
+                        "SERVER",
+                        "STATISTICS",
+                        "SUBSCRIPTION",
+                        "TABLESPACE",
+                        "TRIGGER",
+                        "TYPE",
+                        Sequence("EVENT", "TRIGGER"),
+                        Sequence("FOREIGN", "DATA", "WRAPPER"),
+                        Sequence("FOREIGN", "TABLE"),
+                        Sequence("MATERIALIZED", "VIEW"),
+                        Sequence("TEXT", "SEARCH", "CONFIGURATION"),
+                        Sequence("TEXT", "SEARCH", "DICTIONARY"),
+                        Sequence("TEXT", "SEARCH", "PARSER"),
+                        Sequence("TEXT", "SEARCH", "TEMPLATE"),
+                    ),
```

Comment:
Been thinking this over a bit myself and I’m honestly not sure of the best answer btw, so raised this more to see if you’d thought about it and landed on this solution or hadn’t really thought about it.

On the one hand we have a lot of these more specific identifiers - so why not use them? It could avoid some illegal syntax being accepted as legal (e.g. `COMMENT ON COLUMN Table1 IS '123'` when you need a column name needs to be of the format `table.column`). So going as specific as possible is surely a good thing.

On the other hand, it’s questionable whether this would catch much extra (it wouldn’t catch `COMMENT ON COLUMN Schema1.Table1 IS '123'` for example, nor `COMMENT ON COLUMN noexistent-table.noexistent-column IS '123'`), as you yourself note in your comment it’s not even sure if the type is a fully qualified column, there’s a question as to how much a linter/parser/auto-formatted SHOULD check on this, and - most importantly - it makes the code quite a bit longer and more verbose.

I think I’m veering on the side of “keep it generic” as the gains otherwise are questionable at best. Perhaps the best option is a middle group where we follow the spec and use the same types where it does (e.g. `TABLE`, `VIEW`, `TYPE`), and more specific ones where it does (e.g. `COLUMN`, `FUNCTION`, `CONSTRAINT`… etc)? So a slight collapsing of what you currently have, rather than the fuller collapsing I suggested in code suggestion above?

Or maybe I’m just over thinking this and what you have is absolutely fine? It’s only a small amount of extra code.

WDYT?

Reply:
@tunetheweb Thanks for the review! I'd thought about this a bit during my implementation.
I considered that getting more explicit return types from the parser would prove beneficial beyond the linter aspect, and provide a path to the API being able to provide more in-depth query capabilities on the types of objects returned.
For example, parsing a SQL file and finding "items that are tables" is a valuable operation, and with generic objects, that's not possible.

I was trying to map to existing ReferenceSegments where possible, but if it's preferable to have the more generic object, then I'm happy to reshape that way.

A more generic approach will a likely make implementing the "more exotic" comment syntaxes that are commented (pun!) out in the fixture file a little easier.

> Perhaps the best option is a middle group where we follow the spec and use the same types where it does ...

If I'm reading this correctly, then the total set of potential groups would map closer to the ["Parameters" list in the docs](https://www.postgresql.org/docs/13/sql-comment.html#id-1.9.3.52.6), roughly:

```shellsession
$ sort params.txt | uniq -c
   1 aggregate_name
   2 constraint_name
   1 function_name
   1 large_object_oid
  29 object_name
   1 operator_name
   1 policy_name
   1 procedure_name
   1 relation_name
   1 routine_name
   1 rule_name
   1 source_type
   1 trigger_name
   1 type_name
```

So we would expect the majority of the entries to be as you initially proposed, with carveouts for the more specific params. I'm thinking that structurally this long sequence would be the first entry in the OneOf block, as opposed to last where it is now.

Reply:
The more I think about it, the more I think I was perhaps being a bit pedantic. It's not that much extra code, they are already separate "lines" in the spec so we're still aligning to the spec (even if we're being a bit more specific than they are).

So, after all this, I'm tempted to leave it as is.

WDYT?

Reply:
I'm leaning towards more explicit where possible, like it is now, with the TODO breadcrumb on extracting the more-specific references when they are necessary.

Reply:
I'll also add a couple more sequences and handle some of the input params in another commit.

Reply:
OK resolving this then.

FYI we're having problems with our CircleCI account so don't be surprised when it doesn't run. It's not you, it's us 😞

Reply:
> FYI we're having problems with our CircleCI account so don't be surprised when it doesn't run. It's not you, it's us 😞

Maybe time to migrate to GitHub Actions? 😉